### PR TITLE
MEJO-65 Make different log severities

### PR DESCRIPTION
**Pull request by szveikko**

### CODE REVIEW

File: Nucleo_F746ZG_app/Src/interface_execute.c [hunk 1]
```diff
@@ -204,15 +208,36 @@ void execute_state(command_t* command)
...
+  case LOG_ERROR:
+    log_type = LOG_ERROR;
+    break;
+  case LOG_DEBUG:
+    log_type = LOG_DEBUG;
+    break;
+  case LOG_ALL:
+    log_type = LOG_ALL;
+    break;
+  }
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/interface_execute.c]:
```C
void execute_log(command_t* command)
{
  log_type = command->params.log;
}
```

File: Nucleo_F746ZG_app/Src/utility.c [hunk 2]
```diff
@@ -9,22 +9,29 @@ void utility_init()
...
+    if (messages_data != NULL)
+    {
+      strcpy(messages_data->array, printable_message);
+      osMailPut(mail_box_id, messages_data);
+    }
   }
 }
+
+
+
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/utility.c]: pointless whitespace

File: Nucleo_F746ZG_app/Src/test.c [hunk 3]
```diff
@@ -210,13 +210,13 @@ void TestTask(void const * argument)
...
     osDelay(250);
   }
   rplidar_stop_motor();
   osDelay(1000);
 
-  async_log("Semi-auto tests finished!");
+  async_log(LOG_INFO, "Semi-auto tests finished!");
 
   osThreadTerminate(NULL);
   // semi-automatic test implementation END here
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/test.c]: imho, everything in test.c should be debug. not sure tho, waiting for other opinions

File: Nucleo_F746ZG_app/Src/rplidar_task_utility.c [hunk 4]
```diff
@@ -78,9 +78,9 @@ void LidarTaskHandler(void const *argument)
         }
       }
       if (measurement_error < 3) {
-        async_log("Distance %d - %d: %d\r\n", rplidar_angle_from, rplidar_angle_to, rplidar_get_distance(rplidar_angle_from, rplidar_angle_to));
+        async_log(LOG_INFO, "Distance %d - %d: %d\r\n", rplidar_angle_from, rplidar_angle_to, rplidar_get_distance(rplidar_angle_from, rplidar_angle_to));
       } else {
-        async_log("Measurement failed, please try to reset!\r\n");
+        async_log(LOG_ERROR, "Measurement failed, please try to reset!\r\n");
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/rplidar_task_utility.c]: this is also an info, not an error

File: Nucleo_F746ZG_app/Src/main.c [hunk 5]
```diff
@@ -315,7 +315,7 @@ void HAL_UART_RxCpltCallback(UART_HandleTypeDef *huart)
         if(command.command_status == CMD_STATUS_DONE) {
           execute_command(&command);
         } else {
-          async_log("%s", command.error_message);
+          async_log(LOG_ERROR ,"%s", command.error_message);
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Src/main.c]: this is a warning

File: Nucleo_F746ZG_app/Inc/dynamic_tree.h [hunk 6]
```diff
@@ -12,7 +12,7 @@
 #else
 #include "FreeRTOS.h"
 #include "utility.h"
-#define log(format, data) async_log(format, data)
+#define log(format, data) async_log(LOG_INFO, format, data)
```
Comment (pVer297) [inline Nucleo_F746ZG_app/Inc/dynamic_tree.h]: this is used only for a warning, i just wrote it up so it should be like this

**Comment by pVer297** [comment]
Some kind of severity hierarchy would be nice i think, like if you set warning you still see errors, or if you set info you still see warnings and errors

on the other hand

I don't see too much point in this system, its nice and cool, but what we really use right now is:
<warning - not warning>
"only warnings" or "everything"

File: Nucleo_F746ZG_app/Src/test.c (hunk 3 above)
Comment (lengyeltamas) [reply Nucleo_F746ZG_app/Src/test.c]: I agree with it

File: Nucleo_F746ZG_app/Inc/command_enums.h [hunk 7]
```diff
@@ -171,6 +171,16 @@ typedef enum {
...
 
+/* LOG */
+
+typedef enum {
+  LOG_INFO,
+  LOG_WARNING,
+  LOG_ERROR,
+  LOG_DEBUG,
+  LOG_ALL
+} log_command_t;
```
Comment (johnyka) [inline Nucleo_F746ZG_app/Inc/command_enums.h]: this should be called severity_t

File: Nucleo_F746ZG_app/Inc/utility.h [hunk 8]
```diff
@@ -19,9 +19,11 @@ typedef struct
 osMailQId mail_box_id;
 osThreadId printerThreadHandle;
 
+uint8_t log_type;
 void utility_init();
 void bluetooth_init();
-void async_log(char *format, ...)__attribute__ ((format (printf, 1, 2)));
+void async_log(log_command_t input_log_status, char *format, ...)__attribute__ ((format (printf, 2, 3)));
```
Comment (johnyka) [inline Nucleo_F746ZG_app/Inc/utility.h]: the argument should be called severity

**Comment by szveikko** [comment]
> I'm sorry but this story doesn't make any sense for me. We already have async log and as far as I know we won't have time to create udp/tcp srv on the board and create real time logging for lidar, ultrasonic, rgb, anything related to car++ then we don't need this. By the way that would make sense to store everything that we want as log in the ram on the car and when we call something stupid like "get logs" with a client then the client program could wait for data to be delivered, create a file called something stupid like "car++.log" then delete the data from ram and we could see what went wrong etc. etc.

The changes were applied, so in the future we could displace different kinds of log types. Their use would be handy when debugging f.e.: when something in the car doesnt work, we just get all LOG_DEBUG async logs, and with the help of the acquired info, we can debug succesfully - this also will need more insertions of LOG_DEBUG cases. But the main point of making different severities, is to be able to handle the real time issues of the car, in the way we want. Also the tcp server is not needed for this, since we have bluetooth connection, which can give us the output we need.

**Pull request merged**